### PR TITLE
Aligns Softmax masking behavior with JAX for fully masked axis

### LINKED ISSUE #21123: Softmax layer diverges from jax.nn.softmax
repro colab (10 lines of code): https://colab.research.google.com/drive/1F9-I8Ax-OBKT0-6GmOq5macFBpu-cFid?usp=sharing

The keras and jax version of softmax have a different behavior on data that is completely masked out.

### PR DESCRIPTION
**Pull request by JyotinderSingh**
Fixes #21123

This change modifies the `keras.layers.activations.Softmax` layer to ensure consistent behavior when masking is applied, particularly for the edge case where all elements along the softmax axis are masked.

## Issue

1. The masking mechanism worked by adding a large negative number to the logits corresponding to masked positions before applying the softmax activation.
2. While this effectively minimized the contribution of masked elements when some elements were unmasked, it led to a non-zero, uniform distribution (e.g., `[0.25, 0.25, 0.25, 0.25]`) when all elements along the softmax axis were masked. This behavior differs from the semantics of `jax.nn.softmax(where=mask)`, which outputs zeros for all elements in a fully masked slice.

## Fix

The softmax output tensor is multiplied element-wise by the original boolean mask, ensuring any masked softmax output position is zero'd out.

```python
if mask is not None:
    # Apply the mask to the softmax output to ensure that masked
    # values are set to 0 in case the entire axis is masked.
    outputs = outputs * backend.cast(mask, outputs.dtype)
```

### CODE REVIEW

**Review by fchollet — APPROVED** [review]
LGTM, thanks for the PR!

File: keras/src/layers/activations/softmax.py [hunk 1]
```diff
@@ -58,15 +58,23 @@ def call(self, inputs, mask=None):
...
-                return activations.softmax(inputs, axis=self.axis[0])
-        return activations.softmax(inputs, axis=self.axis)
+                outputs = activations.softmax(inputs, axis=self.axis[0])
+        else:
+            outputs = activations.softmax(inputs, axis=self.axis)
+
+        if mask is not None:
+            # Apply the mask to the softmax output to ensure that masked
+            # values are set to 0 in case the entire axis is masked.
+            outputs = outputs * backend.cast(mask, outputs.dtype)
```
[the remark this answers is not in the document]
Comment (JyotinderSingh) [reply keras/src/layers/activations/softmax.py]: Done
